var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -90,13 +98,26 @@ class Esmf(MakefilePackage):
     depends_on("netcdf-fortran@3.6:", when="+netcdf")
     depends_on("parallel-netcdf@1.2.0:", when="+pnetcdf")
     depends_on("xerces-c@3.1.0:", when="+xerces")
-    depends_on("parallelio@2.5.7: +mpi", when="@8.3.0:8.3.99+external-parallelio+mpi")
-    depends_on("parallelio@2.5.7: ~mpi", when="@8.3.0:8.3.99+external-parallelio~mpi")
-    depends_on("parallelio@2.5.9: +mpi", when="@8.4.0:8.4.99+external-parallelio+mpi")
-    depends_on("parallelio@2.5.9: ~mpi", when="@8.4.0:8.4.99+external-parallelio~mpi")
-    depends_on("parallelio@2.5.10: +mpi", when="@8.5.0:+external-parallelio+mpi")
-    depends_on("parallelio@2.5.10: ~mpi", when="@8.5.0:+external-parallelio~mpi")
-    depends_on("cmake@3.5.2:", type="build", when="~external-parallelio")
+    depends_on(
+        "parallelio@2.5.7: +mpi+pnetcdf", when="@8.3.0:8.3.99+external-parallelio+mpi+pnetcdf"
+    )
+    depends_on("parallelio@2.5.7: +mpi", when="@8.3.0:8.3.99+external-parallelio+mpi~pnetcdf")
+    depends_on(
+        "parallelio@2.5.7: ~mpi+pnetcdf", when="@8.3.0:8.3.99+external-parallelio~mpi+pnetcdf"
+    )
+    depends_on("parallelio@2.5.7: ~mpi", when="@8.3.0:8.3.99+external-parallelio~mpi~pnetcdf")
+    depends_on(
+        "parallelio@2.5.9: +mpi+pnetcdf", when="@8.4.0:8.4.99+external-parallelio+mpi+pnetcdf"
+    )
+    depends_on("parallelio@2.5.9: +mpi", when="@8.4.0:8.4.99+external-parallelio+mpi~pnetcdf")
+    depends_on(
+        "parallelio@2.5.9: ~mpi+pnetcdf", when="@8.4.0:8.4.99+external-parallelio~mpi+pnetcdf"
+    )
+    depends_on("parallelio@2.5.9: ~mpi", when="@8.4.0:8.4.99+external-parallelio~mpi~pnetcdf")
+    depends_on("parallelio@2.5.10: +mpi+pnetcdf", when="@8.5.0:+external-parallelio+mpi+pnetcdf")
```

Comment:
parallelio is now at version 2.6.2 - can we update these or just remove the version so it will get the latest default?

Reply:
The way this is written is that
- for 8.5.0, you need at least 2.5.10, but spack will pick the latest version available unless a specific version is requested
- for 8.4.0-8.4.99, you need at least 2.5.9, but spack will pick the latest version available unless a specific version is requested
- for 8.3.0-8.3.99, you need at least 2.5.7, but spack will pick the latest version available unless a specific version is requested

There is no constraint on the upper version for each of them.

I will note that we should be able to replace `@8.4.0:8.4.99` with just `@8.4` and `@8.3.0:8.3.99` with just `@8.3` (correct @AlexanderRichert-NOAA?)

Reply:
@climbfuji - it looks to me that this section brings back an old version of the parallelio dependency section. It was fixed in #37183, specifically the "Simplify pnetcdf handling wrt parallelio ..." item.

@jedwards4b - this just specifies the "least" parallelio version a specific ESMF version needs. Any more recent parallelio will do, and should be used by default, right?

Reply:
@climbfuji - if `@8.4` will do the trick vs `@8.4.0:8.4.99`, then I am all for simplifying it that way. Thanks!